### ISSUE TITLE
Add support for vendor specific DateParts in DSL.trunc()

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8418]
#2132 introduced vendor specific date parts, which were already supported for `EXTRACT()`. Another useful function to use these date parts with is `TRUNC()`

**Comment by lukaseder** [comment]
Please use plain SQL templating for this case: https://www.jooq.org/doc/latest/manual/sql-building/plain-sql-templating/

**Comment by lukaseder** [comment]
I'm not sure how to emulate `WEEK` in other dialects. Native support in PostgreSQL will be added, of course. Further emulations might be added in the future:

- https://github.com/jOOQ/jOOQ/issues/16100

**Comment by lukaseder** [comment]
Probably some time zone issue... This returns the correct timestamp:

```sql
SELECT timestamp with time zone '0999-04-01 00:00:00.0Z';
```

**Comment by lukaseder** [comment]
I've pushed a first draft of this. The PostgreSQL implementation is trivial, of course, but the emulations will need a bit more tests, specifically, I haven't tested `TIMESTAMPTZ` types yet.

Also, I'll look again into supporting more dialects, probably tomorrow.

**Comment by Petersoj** [comment]
@lukaseder any chance the Postgres implementation for `trunc(DatePart.WEEK)` can be released in the next update?

**Comment by lukaseder** [comment]
> [@lukaseder](https://github.com/lukaseder) any chance the Postgres implementation for `timestampAdd(trunc(DatePart.WEEK))` can be released in the next update?

What is that even supposed to mean.

**Comment by lukaseder** [comment]
Yeah, wrong issue. This is a better one: https://github.com/jOOQ/jOOQ/issues/11425

**Comment by lukaseder** [comment]
Anyway, just use a plain SQL template:
https://www.jooq.org/doc/latest/manual/sql-building/plain-sql/plain-sql-templating-language/
